pkg/proc/proto.go

```diff
@@ -51,5 +51,14 @@ func (r *ProtoReader) ReadPacket() (message.MessageType, []byte, error) {
 	}
 
 	msgType := message.MessageType(data[0])
+
+	if msgType == message.MessageTypeError {
+		errorMessage := message.ErrorMessage{}
+
+		errorMessage.Decode(data)
+
+		return msgType, data, fmt.Errorf("proxy error: %s", errorMessage.Error)
```

Comment:
This is not a proper place for error checking. Responsibility of this func is read msg, not interpret it. Moreover, error could defer in severity. For instance, in PostgreSQL notice, log, info msg are all of type of ErrorMsg. I plan do adapt the same in yproxy.

Reply:
Removed error check